codegen/bin/codegen-ios

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env node
+
+var fs = require('fs');
+var path = require('path');
+var mkdirp = require('mkdirp');
+var semver = require('semver');
+var optimist = require('optimist');
+
+// var generator = require(loopback-sdk-ios-codegen);
+var generator = require('../lib/objc-codegen');
+
+var argv = optimist
+  .usage('Generate iOS Objective-C client code for your LoopBack application.' +
+    '\nUsage:' +
+    '\n    $0 [options] path-to-app ios-client-dir-path' +
+    '\n' +
+    '\nExample:' +
+    '\n    $0 -p XX server/app.js client/ios/gen-src')
+  .describe('p', 'The prefix for generated iOS object names.')
+  .default('p', '')
+  .alias({ p: 'model-name-prefix' })
+  .demand(2)
+  .argv;
+
+var appFile = path.resolve(argv._[0]);
+var clientDir = path.resolve(argv._[1]);
+var modelPrefix = argv['model-name-prefix'];
+
+console.error('Loading LoopBack app %j', appFile);
+var app = require(appFile);
+// assertLoopBackVersion(appFile);
+
+console.error('Generating iOS models with prefix %j', modelPrefix);
+var files = generator.objcModels(app, modelPrefix);
+
+if (!fs.existsSync(clientDir)){
+  mkdirp.mkdirP.sync(clientDir);
+}
+
+for (var objcFileName in files) {
+  fs.writeFileSync(clientDir + "/" + objcFileName, files[objcFileName]);
+  console.log('--- ' + objcFileName);
+  console.log(files[objcFileName]);
+}
+
+// The app.js scaffolded by `slc lb project` loads strong-agent module that
+// used to have a bug where it prevented the application from exiting.
+// To work around that issue, we are explicitly exiting here.
+//
+// The exit is deferred to the next tick in order to prevent the Node bug:
+// https://github.com/joyent/node/issues/3584
+process.nextTick(function() {
+  process.exit();
+});
+
+//--- helpers ---//
+
+function assertLoopBackVersion( appFile ) {
+  var Module = require('module');
+
+  // Load the 'loopback' module in the context of the app.js file,
+  // usually from node_modules/loopback of the project of app.js
+  var loopback = Module._load('loopback', Module._cache[appFile]);
+
+  if (semver.lt(loopback.version, '1.6.0')) {
+    console.error(
+      '\nThe code generator does not support applications based\n' +
+      'on LoopBack versions older than 1.6.0. Please upgrade your\n' +
+      'project to a recent version of LoopBack and run this tool again.\n');
+    process.exit(1);
```

Comment:
Since the script is rather simple, I'm thinking to use a simplified approach for error handling like the following.  Would it look OK with you?

```js
// here
throw new Error('LoopBack version too old\n' +...);

// at the top level of the CLI script
try {
  // all the code generation procedures 
}
catch(err) {
  console.error('\nERROR: ' + err.message + '\n');
}
```